src/Web/Telegram/API/Bot/API.hs

```diff
@@ -69,185 +74,182 @@ newtype Token = Token Text
 telegramBaseUrl :: BaseUrl
 telegramBaseUrl = BaseUrl Https "api.telegram.org" 443 ""
 
--- | Type for token
-type TelegramToken = Capture ":token" Token
-
 -- | Telegram Bot API
 type TelegramBotAPI =
```

Comment:
Situation is slightly worse due to manager:

```
sendMessage token alice "hello" manager
sendMessage token bob "hello" manager
sendMessage token charlie "hello" manager
```

I would definitely go with solution proposed by you with monadic telegram client.